remote-api/src/main/java/org/alfresco/repo/web/scripts/servlet/RemoteUserAuthenticatorFactory.java

```diff
@@ -181,7 +181,7 @@ public boolean authenticate(RequiredAuthentication required, boolean isGuest)
                         return false;
                     }
                 }
-                else if (userId == null)
+                if (userId == null)
```

Comment:
I meant, `if (userId == null)` should wrap this whole section L166 to L184 and beyond as this is now a precondition for both. See the repetition:

`if (isAlwaysAllowBasicAuthForAdminConsole() && userId == null)`
...
`else/if (userId == null)`